### ISSUE TITLE
Numbers get parsed as dates

### ISSUE THREAD

**Issue by lambrojos** [DaSpawn/xlsx-stream-reader#21]
Hi,
I am experiencing an issue in which I try to parse a file containing both a date column and a five digit number (italian zipcodes). The date gets parsed correctly but the number gets parsed as a date as well!

I can provide a failing test case here:
https://github.com/lambrojos/xlsx-stream-reader/blob/number-formatted-as-date/test/test.js#L20

Thanks!

**Comment by lambrojos** [comment]
Sorry I had committed a wrong test case. The issue however, remains

**Comment by DaSpawn** [comment]
Do you have a file with an example?

**Comment by lambrojos** [comment]
The branch on my fork containg the failing tests also includes an example file: https://github.com/lambrojos/xlsx-stream-reader/blob/number-formatted-as-date/test/import.xlsx

**Comment by lambrojos** [comment]
However I think I have tracked the issue..
Here
https://github.com/DaSpawn/xlsx-stream-reader/blob/master/lib/worksheet.js#L246
the library just checks for the existance of the format id, but it does note verify it is a date format..

For example in my test, `format id` contains:
```
// actual date
attributes: 
   {
 numFmtId: '166',
 ...
} 

// number
 { 
numFmtId: '164',
...
   }
```

while `numFts` contains

```
 [ { name: 'numFmt',
    attributes: { numFmtId: '164', formatCode: 'General' } },
  { name: 'numFmt',
    attributes: { numFmtId: '165', formatCode: '@' } },
  { name: 'numFmt',
    attributes: { numFmtId: '166', formatCode: 'DD/MM/YYYY' } },
  { name: 'cellStyleXfs', attributes: { count: '20' } } ]
```
If we look up formats by  number format id it is evident that the number column (numFmtId = 164) matches with the ''General' format code  while the date (numFmtID = 166) matches with the DD/MM/YYYY format.

If that makes sense to you, I will provide a pr implementing this check. Moreover it would be nice to use https://www.npmjs.com/package/ssf to parse formats... what do you think? Here is a demo http://oss.sheetjs.com/ssf/

**Comment by DaSpawn** [comment]
Looks like this was a lazy fix on my part to handle dates in my sheets at the time (most of my working sheets were inline cell data or shared string).  The SSF package looks like a good candidate/fix to handle that case of field formatting and would be much more compatible for all the various formats

**Comment by lambrojos** [comment]
https://github.com/DaSpawn/xlsx-stream-reader/pull/22

**Comment by DaSpawn** [comment]
fixed with #22
